src/PointwiseFunctions/AnalyticSolutions/NewtonianEuler/LaneEmdenStar.cpp

```diff
@@ -29,6 +29,59 @@ void LaneEmdenStar::pup(PUP::er& p) noexcept {
   p | central_mass_density_;
   p | polytropic_constant_;
   p | equation_of_state_;
+  p | source_term_;
+}
+
+template <>
+tnsr::I<double, 3> LaneEmdenStar::gravitational_field(
+    const tnsr::I<double, 3>& x) const noexcept {
+  // Compute alpha for polytrope n==1, units G==1
+  const double alpha = sqrt(0.5 * polytropic_constant_ / M_PI);
+  const double outer_radius = alpha * M_PI;
+  const double mass_scale = 4.0 * M_PI * cube(alpha) * central_mass_density_;
+  // Add tiny offset to avoid divisons by zero
+  const double radius = get(magnitude(x)) + 1.e-30 * outer_radius;
+
+  double enclosed_mass = mass_scale;
+  if (radius < outer_radius) {
+    const double xi = radius / alpha;
+    enclosed_mass *= sin(xi) - xi * cos(xi);
+  } else {
+    enclosed_mass *= M_PI;
+  }
+
+  auto gravitational_field_result = x;
+  for (size_t i = 0; i < 3; ++i) {
+    gravitational_field_result.get(i) *= -enclosed_mass / cube(radius);
+  }
+  return gravitational_field_result;
+}
+
+template <>
+tnsr::I<DataVector, 3> LaneEmdenStar::gravitational_field(
```

Comment:
You might also want to use `get_size` and `get_element` (see commit on solution) here if you want to save some lines. Up to you.

Reply:
I also think that would be good.